src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -499,23 +545,17 @@ public void add(final String moduleName, final ExternalCommandlineProgramModule.
             public String getId() { return id; }
         }
 
-        private List<Tuple2<ContigID, ContigSequence>> contents;
+        private final List<Tuple2<ContigID, ContigSequence>> contents;
 
         public List<Tuple2<ContigID, ContigSequence>> getContents(){
             return contents;
         }
 
-        public ContigsCollection(final File fastaFile) throws IOException{
-
-            if(null==fastaFile){// early return if the fastaFile is null, which signals something went wrong in sga pipeline
-                throw new IOException(fastaFile.getName() + "is null");
-            }
-
-            final List<String> lines = Files.readAllLines(Paths.get(fastaFile.getAbsolutePath()));
+        public ContigsCollection(final List<String> fileContents){
 
             contents = new ArrayList<>();
-            for(int i=0; i<lines.size(); i+=2){
-                contents.add(new Tuple2<>(new ContigID(lines.get(i)), new ContigSequence(lines.get(i+1))));
+            for(int i=0; i<fileContents.size(); i+=2){
```

Comment:
I meant to pull the file reading and parsing out to the calling method.. The fact that the lines are interleaved ID/Sequence is really a file format thing, so I'd prefer to move this code out to where you pass the file and instead pass in a list of <ID,Seq> tuples. Not a big deal but it seems like a better separation to me.